src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,61 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FilterBy } from '../../types/FilterBy';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  setFilterBy: (filter: FilterBy) => void;
+  setQuery: (query: string) => void;
+  query: string;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  setFilterBy,
+  setQuery,
+  query,
+}) => {
+  const handleResetQuery = () => setQuery('');
+
+  const handleSetQuery = (event: React.ChangeEvent<HTMLInputElement>) =>
+    setQuery(event.target.value);
+
+  const handleFilterBy = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setFilterBy(event.target.value as FilterBy);
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleFilterBy}>
+            <option value={FilterBy.All}>All</option>
+            <option value={FilterBy.Active}>Active</option>
+            <option value={FilterBy.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={handleSetQuery}
+        />
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
- Don't use inline styles, use classes instead
- Don't disable linter

Reply:
The linter was disabled in the initial code, I thought it was better not to touch it.

Reply:
the same situation with inline styles.